liquidity-pool-changes/src/agent.js

```diff
@@ -0,0 +1,219 @@
+const BigNumber = require('bignumber.js'); // always convert ethers.js bignumber to javascript bignumber
+const axios = require('axios');
+
+const {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  ethers,
+  getEthersProvider,
+} = require('forta-agent');
+
+// load agent configurations
+const config = require('../agent-config.json');
+
+const utils = require('./utils');
+
+const DECIMALS_ABI = ['function decimals() view returns (uint8)'];
+
+// used to store initialization data once you pass it into the provideHandleBlock() function
+const initializeData = {};
+
+function provideInitialize(data) {
+  return async function initialize() {
+    /* eslint-disable no-param-reassign */
+    // setup ethers.js provider to interact with contracts
+    data.provider = getEthersProvider();
+
+    // get abi
+    data.erc20Abi = utils.getAbi('ERC20.json');
+
+    // get liquidityThresholdPercentChange
+    data.liquidityThresholdPercentChange = new BigNumber(
+      config.liquidityThresholdPercentChange,
+    );
+    /* eslint-disable no-param-reassign */
+  };
+}
+
+async function getTokenPrices(token0Address, token1Address) {
+  const coingeckoApiUrl = 'https://api.coingecko.com/api/v3/simple/token_price/ethereum?';
+  const addressQuery = `contract_addresses=${token0Address},${token1Address}`;
+  const vsCurrency = '&vs_currencies=usd';
+
+  const url = coingeckoApiUrl.concat(addressQuery.concat(vsCurrency));
+  const { data } = await axios.get(url);
+
+  // parse response to convert to BigNumber.js
+  const usdPerToken0 = new BigNumber(data[token0Address.toLowerCase()].usd);
+  const usdPerToken1 = new BigNumber(data[token1Address.toLowerCase()].usd);
+
+  return { token0Price: usdPerToken0, token1Price: usdPerToken1 };
+}
+
+// amountBN is the amount of tokens in the liquidity pool in BigNumber.js format
+async function getTokenUSDValue(amountBN, tokenPrice, tokenAddress, provider) {
+  // get the decimal scaling for this token
+  const contract = new ethers.Contract(tokenAddress, DECIMALS_ABI, provider);
+  let decimals;
+  try {
+    // calling .decimals() may fail for a vyper contract
+    decimals = await contract.decimals();
+  } catch {
+    return undefined;
+  }
+  const denominator = new BigNumber(10).pow(decimals);
+
+  // multiply by the price and divide out decimal places
+  return amountBN.times(tokenPrice).div(denominator);
+}
+
+let previousLiquidity; // keep state of liquidity during previous block
+let counter = 0; // instantiate previousLiquidity amount the first time this agent is ran
```

Comment:
Couple of things here:

1. It would be better if `previousLiquidity` and `counter` were instantiated under the data object provided to the `initialitze` function instead of having these variables be in global scope. Additionally, then you wouldn't need the `getPreviousLiquidity` getter function for testing.
2. The variable named `counter` seems like a misnomer as it's more of a flag to check if `previousLiquidity` has been instantiated - it could also be removed in favor of checking if `previousLiquidity` is undefined.

Reply:
this makes sense, I was able to instantiate it under the data object and get rid of the `counter` variable as well by checking if `previousLiquidity` is undefined